sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeletedKey.cs

```diff
@@ -15,6 +15,7 @@ public class DeletedKey : Key
 
         public DateTimeOffset? ScheduledPurgeDate { get; private set; }
 
+        public DeletedKey() { }
```

Comment:
Should we consider making this `protected` instead of `public`?   It seems like it may be intended for mocking but otherwise has no way to set property values.

_(side note: if so, we may want to consider the `private` settings as `protected`)_

Reply:
This type is actually returned by the service when doing a DeleteKey call. It is true that the user can't set the values but can interact with them if needed. With that we decided public, but if there is a better way I am open to it

Reply:
Apologies for the confusion;  I was just referring to the empty constructor, not the type.   I see how that may have been unclear.

Reply:
Ohh I see. Well, we need it for the `GetDeletedKeysAsync` API where it return an enumerator of `DeletedKeys` and needs an empty constructor https://github.com/Azure/azure-sdk-for-net/pull/6479/files#diff-70b613409e65d848de58b28efa0d2d22R174